projects/plugins/social/src/class-share-limits.php

```diff
@@ -28,21 +28,34 @@ class Share_Limits {
 	 */
 	public $shares_remaining;
 
+	/**
+	 * If the blog has a Jetpack Social paid plan.
+	 *
+	 * @var boolean
+	 */
+	public $has_paid_plan;
+
 	/**
 	 * Constructor.
 	 *
-	 * @param array $connections List of Publicize connections.
-	 * @param int   $shares_remaining Number of shares remaining for this period.
+	 * @param array   $connections List of Publicize connections.
+	 * @param int     $shares_remaining Number of shares remaining for this period.
+	 * @param boolean $has_paid_plan If the blog has a Jetpack Social paid plan.
 	 */
-	public function __construct( $connections, $shares_remaining ) {
+	public function __construct( $connections, $shares_remaining, $has_paid_plan ) {
 		$this->connections      = $connections;
 		$this->shares_remaining = $shares_remaining;
+		$this->has_paid_plan    = $has_paid_plan;
 	}
 
 	/**
 	 * Run functionality required to enforce sharing limits.
 	 */
 	public function enforce_share_limits() {
+
+		if ( $this->has_paid_plan ) {
+			return;
+		}
```

Comment:
Is there a reason you added this to the class itself, and not the function where the class gets called (in `class-jetpack-social.php`)? That's where the other checks happen before the class gets instantiated.

Reply:
Good idea, I will do that!